lib/logstash/event.rb

```diff
@@ -296,6 +296,7 @@ def to_json_with_metadata(*args)
   def self.validate_value(value)
     case value
     when String
+      value.force_encoding("UTF-8") if LogStash::Environment.windows?
```

Comment:
This solution feels a bit blunt. We're targeting `Socket.gethostname` and `Socket#peeraddr` (and maybe others).

* On MRI 2.1.4 and OSX, `Socket.gethostname` and `Socket#peeraddr` return strings that are ASCII-8BIT on my OSX laptop. This is a bug in MRI.
* On Jruby 1.7.17 and OSX, `Socket.gethostname` returns UTF-8 string
* On JRuby 1.7.17 and OSX, `Socket#peeraddr[3]` is UTF-8

I will test on Windows shortly.